res/controllers/Denon-MC7000-scripts.js

```diff
@@ -464,59 +472,27 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
-            engine.setValue(group, "beatlooproll_0.125_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x15, MC7000.padColor.rollon);
-        } else if (control === 0x15 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_0.125_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x15, MC7000.padColor.rolloff);
-        } else if (control === 0x16 && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_0.25_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x16, MC7000.padColor.rollon);
-        } else if (control === 0x16 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_0.25_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x16, MC7000.padColor.rolloff);
-        } else if (control === 0x17 && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_0.5_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x17, MC7000.padColor.rollon);
-        } else if (control === 0x17 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_0.5_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x17, MC7000.padColor.rolloff);
-        } else if (control === 0x18 && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_1_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x18, MC7000.padColor.rollon);
-        } else if (control === 0x18 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_1_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x18, MC7000.padColor.rolloff);
-        } else if (control === 0x19 && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_2_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x19, MC7000.padColor.rollon);
-        } else if (control === 0x19 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_2_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x19, MC7000.padColor.rolloff);
-        } else if (control === 0x1A && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_4_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1A, MC7000.padColor.rollon);
-        } else if (control === 0x1A && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_4_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1A, MC7000.padColor.rolloff);
-        } else if (control === 0x1B && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_8_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1B, MC7000.padColor.rollon);
-        } else if (control === 0x1B && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_8_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1B, MC7000.padColor.rolloff);
+        for (i = 0; i <= 7; i++) {
```

Comment:
You can get rid of the for loop and the control value check by calculating the correct value directly:

```suggestion
        i = control - 0x14;
```